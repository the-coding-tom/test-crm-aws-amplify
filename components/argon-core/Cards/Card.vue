<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
