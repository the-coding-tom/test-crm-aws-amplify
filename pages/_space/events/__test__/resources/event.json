{
  "id": "8880f445-122a-45bb-9476-a615d7492881",
  "name": "SHACK15 CONVERSATIONS: LISA KRISTINE",
  "description": "A CONVERSATION AND GUIDED ART GALLERY TOUR WITH HUMANITARIAN PHOTOGRAPHER LISA KRISTINE. Join us for a SHACK15 Conversation with Lisa Kristine, an internationally acclaimed humanitarian photographer and social entrepreneur who has documented indigenous peoples and social causes in more than 150 countries across six continents.",
  "banner_url": "https://d3mym26vamg9ol.cloudfront.net/event/1593010166_d2b06fae-7f1c-43fc-b9d3-1f7ec9493beb.png",
  "host_logo": "https://d3mym26vamg9ol.cloudfront.net/event/1593010176_d743ff8d-28c1-4636-8d6d-f2a8140599a2.png",
  "host_name": null,
  "type": "event",
  "start_time": "2020-06-30 14:00:00",
  "end_time": "2020-06-30 15:00:00",
  "price": 10,
  "total_tickets": 10,
  "max_ticket_per_person": 3,
  "send_email": false,
  "email_subject": "Booking Confirmed",
  "email_content": null,
  "attendees_count": 0,
  "event_category": {
    "id": "8e42f835-74b2-475f-b7f3-cf72c6b792ed",
    "name": "talks",
    "color": "#83ac0c",
    "disable": false,
    "created_at": "2019-11-05 08:49:10",
    "updated_at": "2019-11-05 08:49:10"
  },
  "room": null,
  "external_location": "Lounge",
  "related_attendees": [],
  "banner_image": "https://d3mym26vamg9ol.cloudfront.net/event/1593010170_53b0a7de-b4b5-4018-82cb-4468088d4001.png"
}
