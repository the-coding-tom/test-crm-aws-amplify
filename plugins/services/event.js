export default function(ctx, inject) {
  // const subdomain() = ctx.route.params.space
  const subdomain = () => {
    return ctx.route.params.space
  }

  const Event = {
    addEvent: payload => {
      return ctx.$axios.$post(`/${subdomain()}/events`, payload)
    },
    deleteEvent: eventId => {
      return ctx.$axios.$delete(`/${subdomain()}/events/${eventId}`)
    },
    updateEvent: (id, payload) => {
      return ctx.$axios.put(`/${subdomain()}/events/${id}`, payload)
    },
    getEvent: eventId => {
      return ctx.$axios.$get(`/${subdomain()}/events/${eventId}`)
    },
    getEvents() {
      return ctx.$axios.$get(`/${subdomain()}/events`)
    },
    addCategory(payload) {
      return ctx.$axios.$post(`/${subdomain()}/event-categories`, payload)
    },
    updateCategory(eventId, payload) {
      return ctx.$axios.$patch(
        `/${subdomain()}/event-categories/${eventId}`,
        payload
      )
    },
    getEventCategories() {
      return ctx.$axios.$get(`/${subdomain()}/event-categories`)
    },
    deleteCategory(eventId) {
      return ctx.$axios.delete(`/${subdomain()}/event-categories/${eventId}`)
    },
    getRooms() {
      return ctx.$axios.get(`/${subdomain()}/admin/rooms`)
    },
    getAttendees(eventId = null, link = null) {
      if (!eventId) {
        return ctx.$axios.get(link)
      }

      return ctx.$axios.get(`/${subdomain()}/events/${eventId}/attendees`)
    },
    purchaseTicketByAdmin(eventId, payload) {
      return ctx.$axios.post(
        `/${subdomain()}/events/${eventId}/purchase-ticket`,
        payload
      )
    },
    cancelTicket(eventId, payload) {
      return ctx.$axios.delete(
        `/${subdomain()}/events/${eventId}/cancel-ticket`,
        payload
      )
    },
    getMemberships() {
      return ctx.$axios.get(`/${subdomain()}/memberships`)
    },
    getEventsByDate(startDate, endDate) {
      return ctx.$axios.get(
        `/${subdomain()}/events?filter[starts_between]=${startDate},${endDate}`
      )
    },
    deleteEvent(eventId) {
      return ctx.$axios.delete(`/${subdomain()}/events/${eventId}/`)
    }
  }
  ;(ctx.$event = Event), inject('event', Event)
}
