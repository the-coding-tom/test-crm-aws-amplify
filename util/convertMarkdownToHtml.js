export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
