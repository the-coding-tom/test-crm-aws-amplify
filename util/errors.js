export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
