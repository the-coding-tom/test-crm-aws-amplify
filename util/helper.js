const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
