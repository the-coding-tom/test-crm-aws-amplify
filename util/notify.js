const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
