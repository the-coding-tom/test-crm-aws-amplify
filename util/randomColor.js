const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
