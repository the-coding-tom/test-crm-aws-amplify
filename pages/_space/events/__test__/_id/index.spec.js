import IdIndexPage from '@/pages/_space/events/_id/index.vue'
import IdEditPage from '@/pages/_space/events/_id/edit.vue'
import IndexPage from '@/pages/_space/events/index.vue'
import Card from '~/components/argon-core/Cards/Card.vue'
import Vue2Filters from 'vue2-filters'
import BootstrapVue from 'bootstrap-vue'
import Badge from '~/components/argon-core/Badge.vue'

import VueRouter from 'vue-router'
import { createWrapper } from '@/test.util.js'
import moment from 'moment'

import event from '../resources/event.json'
import currentSpace from '@/pages/_space/__test__/resources/current-space.json'

describe('Page: /:space/events/:id | Events Id Index Page', () => {
  let NuxtStore, store, wrapper, $event, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $event = {
      getEvent: jest.fn().mockResolvedValue({ data: event })
    }
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()
    store.commit('space/setSpace', currentSpace)

    router = new VueRouter({
      routes: [
        {
          name: 'space-events',
          path: '_space/events',
          component: IndexPage
        },
        {
          name: 'space-events-id-edit',
          path: '_sapce/events/:id/edit',
          component: IdEditPage
        }
      ]
    })

    const params = {
      id: event.id
    }

    const asyncDataResponse = await IdIndexPage.asyncData({
      store,
      $event,
      params
    })

    wrapper = createWrapper(
      IdIndexPage,
      store,
      router,
      {
        ...asyncDataResponse
      },
      {
        stubs: {
          'nuxt-link': true,
          'client-only': true,
          transition: false
        }
      },
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.use(Vue2Filters)
        vueInstance.component(Card.name, Card)
        vueInstance.component(Badge.name, Badge)
        vueInstance.prototype.$event = $event
        vueInstance.prototype.$moment = moment
      }
    )
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
