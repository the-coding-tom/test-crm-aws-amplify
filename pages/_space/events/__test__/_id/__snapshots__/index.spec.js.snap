// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/events/:id | Events Id Index Page renders properly 1`] = `
<div>
  <div class="header pb-6">
    <div class="container-fluid">
      <div class="header-body">
        <div class="d-flex justify-content-between align-items-center py-4">
          <h1 class="page-title"><span>Preview Event</span> <b style="display: none;">â€¢</b> </h1> <a href="#_sapce/events/8880f445-122a-45bb-9476-a615d7492881/edit" class="btn btn-primary" target="_self"><i class="ti-pencil"></i> Edit Event
          </a>
        </div>
      </div>
    </div>
  </div>
  <div class="container-fluid mt--6">
    <div class="row">
      <div class="col-md-4">
        <div class="card sh-eventpv">
          <!---->
          <div class="card-body">
            <div class="m-n25 img-wrap"><img src="https://d3mym26vamg9ol.cloudfront.net/event/1593010166_d2b06fae-7f1c-43fc-b9d3-1f7ec9493beb.png" alt="" class="img-fluid"></div>
            <div class="pv-content">
              <h3>SHACK15 CONVERSATIONS: LISA KRISTINE</h3> <span><p>A CONVERSATION AND GUIDED ART GALLERY TOUR WITH HUMANITARIAN PHOTOGRAPHER LISA KRISTINE. Join us for a SHACK15 Conversation with Lisa Kristine, an internationally acclaimed humanitarian photographer and social entrepreneur who has documented indigenous peoples and social causes in more than 150 countries across six continents.</p></span> <span class="badge badge-info">talks</span>
            </div>
            <div class="pv-footer">
              <p>
                Price: $10.00 /
                <span class="text-muted">ticket</span></p>
              <p>June 30 2020, 14:00 - 15:00 </p>
              <div>
                <p>Lounge</p>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
        <div class="mr-b-30"><a href="#_space/events" class="btn text-primary btn-transparent" target="_self"><i class="fa fa-angle-left mr-r-10"></i> Back
          </a></div>
      </div>
    </div>
  </div>
</div>
`;
